In [1]:
# Importing Libraries

import numpy as np
import matplotlib as plt
import seaborn as sns
import pandas as pd
from scipy.stats import ttest_rel
from scipy.stats import ttest_ind
from statsmodels.stats.weightstats import ztest

In [2]:
# Reading the dataset

data= pd.read_csv('dataset w7.csv')
data

Gender Test preparation  Total Marks
0     male             none           14
1   female             none           28
2   female             none           18
3   female             none           48
4   female             none           21
5   female        completed           40
6     male             none           30
7   female             none           12
8     male             none           18
9     male             none           24
10    male        completed           44
11    male             none           23
12    male        completed           43
13    male             none           28
14  female             none           41
15  female             none           13
16  female        completed           48
17  female             none           45
18    male             none           47
19    male             none           36
20    male             none           14
21    male        completed           44
22  female             none           41
23  female             none           27
24  female             none           27
25  female             none           38
26  female        completed           50
27  female        completed           43

In [3]:
# Checking for null values

data.isnull().sum()

Gender              0
Test preparation    0
Total Marks         0
dtype: int64

In [4]:
# 1. Carry out descriptive analytics on the dataset.

data[('Total Marks')].describe()

count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000
Name: Total Marks, dtype: float64

In [5]:
data[('Test preparation')].describe()

count       28
unique       2
top       none
freq        21
Name: Test preparation, dtype: object

In [6]:
data['Gender'].value_counts()

female    16
male      12
Name: Gender, dtype: int64

In [7]:
data.groupby('Gender')['Test preparation'].value_counts()

Gender  Test preparation
female  none                12
        completed            4
male    none                 9
        completed            3
Name: Test preparation, dtype: int64

In [8]:
data.groupby('Gender')['Total Marks'].mean()


Gender
female    33.750000
male      30.416667
Name: Total Marks, dtype: float64

In [9]:
data.groupby('Test preparation')['Total Marks'].mean()

Test preparation
completed    44.571429
none         28.238095
Name: Total Marks, dtype: float64

Inferences: 
1. There are no null values. The total sample size is 28.
2. Out of the total 28 samples, 21 samples didnot take test preparation, 7 completed test preparation and out of which 4 are female.
3. There 16 female and 12 male saqmples
4. The average total marks for female are greater than male samples.
5. The average total marks for those who completed test preparation are greater than for those who didn't take test preparation.
6. The mean score of the total marks is 32.32, where the minimum scored value is 12 and maximum is 50. 

In [10]:
# 2. Check whether female and male students scored the same marks.
#H0:Male and female students scored same marks 
#H1:Male and female students doesn't score same marks

alpha=.05
MM = data[data["Gender"]=="male"][["Total Marks"]]
FM = data[data["Gender"]=="female"][["Total Marks"]]
t_score,p_value=ttest_ind(MM,FM)
print('The test result is:', t_score, p_value)

The test result is: [-0.69408853] [0.49378223]


In [11]:
if (p_value>alpha):
    print('The p value is', p_value,'is greater than 0.05(the alpha value), therefore the dataset fails to reject the null hypothesis.')
else:
    print('The p value',p_value,'is less than 0.05(the alpha value), therefore the sample fails to accept the null hypothesis.')

The p value is [0.49378223] is greater than 0.05(the alpha value), therefore the dataset fails to reject the null hypothesis.


According to the null hypothesis, the female and male candidates scores same marks. 

In [12]:
# 3. Whether test preparation helps the students?
# H0= There is no significant difference between the scores of students who completed test preparation and who didn't.
# H1= There is significant difference between the scores of students who completed test preparation and who didn't.

alpha=.05
CM = data[data["Test preparation"]=="completed"][["Total Marks"]]
NM = data[data['Test preparation']=="none"][["Total Marks"]]
t_score2,p_value2=ttest_ind(CM,NM)
print('The test result is:', t_score2, p_value2)

The test result is: [3.61436375] [0.00126713]


In [13]:
if (p_value2>alpha):
    print('The p value is', p_value2,'is greater than 0.05(the alpha value), therefore the dataset fails to reject the null hypothesis.')
else:
    print('The p value',p_value2,'is less than 0.05(the alpha value), therefore the sample fails to accept the null hypothesis.')

The p value [0.00126713] is less than 0.05(the alpha value), therefore the sample fails to accept the null hypothesis.


According to the alternative hypothesis, there is significant difference between the total marks of students who took test preparation who didn't. Therefore, we can conclude that the test preparation helped the students. 